bigtable-hbase-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java

```diff
@@ -817,8 +820,15 @@ public boolean isTableAvailable(TableName tableName, byte[][] splitKeys) throws
   public void snapshot(String snapshotName, TableName tableName)
       throws IOException, SnapshotCreationException, IllegalArgumentException {
 
-    connection.getSession().getInstanceAdminClient()
-        .waitForOperation(snapshotTable(snapshotName, tableName));
+    try {
+      connection.getSession().getInstanceAdminClient()
+          .waitForOperation(snapshotTable(snapshotName, tableName));
+    } catch (TimeoutException e) {
+      throw new IOException("Timed out waiting for snapshot creation to finish", e);
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new IOException("Interrupted while waiting for snapshot creation to finish");
```

Comment:
I don't know if this is a good idea.
We are getting an InterruptedException from "waitForOperation(snapshotTable(snapshotName, tableName))"

How can we know for sure that the thread is interrupted while waiting for snapshot creation to finish? For all we know, it could be at time when we trying to create the snapshot.